compiler/noirc_evaluator/src/ssa/opt/die/prune_dead_parameters.rs

```diff
@@ -39,43 +39,31 @@
 //! b1(v0: Field):
 //!   return v0
 //! ```
-use fxhash::FxHashSet as HashSet;
-
-use crate::ssa::{
-    ir::{
-        cfg::ControlFlowGraph, function::Function, instruction::TerminatorInstruction,
-        post_order::PostOrder,
-    },
-    ssa_gen::Ssa,
-};
+use fxhash::{FxHashMap as HashMap, FxHashSet as HashSet};
 
-impl Ssa {
-    /// See [`prune_dead_parameters`][self] module for more information.
-    #[tracing::instrument(level = "trace", skip(self))]
-    pub(crate) fn prune_dead_parameters(mut self) -> Self {
-        for function in self.functions.values_mut() {
-            function.prune_dead_parameters();
-        }
-        self
-    }
-}
+use crate::ssa::ir::{
+    basic_block::BasicBlockId, cfg::ControlFlowGraph, function::Function,
+    instruction::TerminatorInstruction, post_order::PostOrder, value::ValueId,
+};
 
 impl Function {
     /// See [`prune_dead_parameters`][self] module for more information
-    pub(crate) fn prune_dead_parameters(&mut self) {
+    pub(crate) fn prune_dead_parameters(
+        &mut self,
+        unused_params: HashMap<BasicBlockId, Vec<ValueId>>,
+    ) {
         let cfg = ControlFlowGraph::with_function(self);
         let post_order = PostOrder::with_cfg(&cfg);
 
-        for block in post_order.as_slice() {
-            let block = *block;
-
-            let unused_params = self.dfg[block].unused_parameters().to_vec();
+        for &block in post_order.as_slice() {
+            let empty_params = Vec::new();
+            let unused_params = unused_params.get(&block).unwrap_or(&empty_params);
             if unused_params.is_empty() {
                 // Nothing to do if the block has no unused params
                 continue;
             }
 
-            self.dfg[block].clear_unused_parameters();
+            // self.dfg[block].clear_unused_parameters();
```

Comment:
```suggestion
```

Reply:
Still making some changes, so I removed this locally